internal/api/models/invariant.go

```diff
@@ -7,56 +7,88 @@ import (
 	NotOK InvResponseStatus = "NOTOK"
 )
 
+// InvRequestParams ... Request params for invariant operation
 type InvRequestParams struct {
-	Network core.Network       `json:"network"`
-	PType   core.PipelineType  `json:"pipeline_type"`
-	InvType core.InvariantType `json:"type"`
+	Network string `json:"network"`
+	PType   string `json:"pipeline_type"`
+	InvType string `json:"type"`
 
 	StartHeight *big.Int `json:"start_height"`
 	EndHeight   *big.Int `json:"end_height"`
 
-	SessionParams interface{} `json:"invariant_params"`
+	SessionParams map[string]interface{} `json:"invariant_params"`
+	AlertingDest  string                 `json:"alert_destination"`
 }
 
-type InvRequestBody struct {
-	Method InvariantMethod  `json:"method"`
-	Params InvRequestParams `json:"params"`
+// AlertingDestType ... Returns the alerting destination type
+func (irp *InvRequestParams) AlertingDestType() core.AlertDestination {
+	return core.StringToAlertingDestType(irp.AlertingDest)
 }
 
-type InvResponse struct {
-	Code   int               `json:"status_code"`
-	Status InvResponseStatus `json:"status"`
+// NetworkType ... Returns the network type
+func (irp *InvRequestParams) NetworkType() core.Network {
+	return core.StringToNetwork(irp.Network)
+}
 
-	Result InvResult `json:"result"`
-	Error  string    `json:"error"`
+// PiplineType ... Returns the pipeline type
+func (irp *InvRequestParams) PiplineType() core.PipelineType {
+	return core.StringToPipelineType(irp.PType)
 }
 
-type InvResult = map[string]string
+// InvariantType ... Returns the invariant type
+func (irp *InvRequestParams) InvariantType() core.InvariantType {
+	return core.StringToInvariantType(irp.InvType)
+}
 
-func (params *InvRequestParams) GeneratePipelineConfig(endpoint string, pollInterval time.Duration,
+// GeneratePipelineConfig ... Generates a pipeline config using the request params
+func (irp *InvRequestParams) GeneratePipelineConfig(endpoint string, pollInterval time.Duration,
 	regType core.RegisterType) *core.PipelineConfig {
 	return &core.PipelineConfig{
-		Network:      params.Network,
+		Network:      irp.NetworkType(),
 		DataType:     regType,
-		PipelineType: params.PType,
+		PipelineType: irp.PiplineType(),
 		OracleCfg: &core.OracleConfig{
 			RPCEndpoint:  endpoint,
 			PollInterval: pollInterval,
-			StartHeight:  params.StartHeight,
-			EndHeight:    params.EndHeight,
+			StartHeight:  irp.StartHeight,
+			EndHeight:    irp.EndHeight,
 		},
 	}
 }
+
+// InvRequestBody ... Request body for invariant operation request
+type InvRequestBody struct {
+	Method InvariantMethod  `json:"method"`
+	Params InvRequestParams `json:"params"`
+}
+
+// InvResult ... Result of invariant operation
+type InvResult = map[string]string
+
+// InvResponse ... Response for invariant operation request
+type InvResponse struct {
+	Code   int               `json:"status_code"`
+	Status InvResponseStatus `json:"status"`
+
+	Result InvResult `json:"result"`
+	Error  string    `json:"error"`
+}
```

Comment:
Are we not planning to use gRPC for creating and maintaining rpc endpoints? Defining endpoints via protobufs could help with strictly typing endpoint request/response models as well as autogenerating code like this

Reply:
Agreed but think that's ultimately out of scope for now